bookkeeper-common/src/main/java/org/apache/bookkeeper/common/util/SingleThreadExecutor.java

```diff
@@ -213,18 +233,30 @@ public void execute(Runnable r) {
                 queue.put(r);
                 tasksCount.increment();
             } else {
-                if (queue.offer(r)) {
-                    tasksCount.increment();
+                int delta = r != null ? 1 : runnableList.size();
+                validateQueueCapacity(delta);
```

Comment:
<img width="466" alt="image" src="https://github.com/user-attachments/assets/dfa66899-b655-4da4-af37-ec50c5d0bc76" />
If a null is passed in here, will runnableList.size() report a NullPointerException? This may affect the function of public void execute(Runnable r)

Reply:
Done.